### PR TITLE
vivado: update for SC 0.10

### PR DESCRIPTION
**Pull request by nmoroze**
Changes verified locally by running `examples/blinky/run_vivado.sh`.

### CODE REVIEW

File: siliconcompiler/tools/vivado/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .vivado import *
```
Comment (gadfort) [inline siliconcompiler/tools/vivado/__init__.py:1]: I see the reason for this, but I think you should do explicit imports so it's clear what exactly it is you want

File: siliconcompiler/tools/vivado/route.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from siliconcompiler.tools import vivado
+
+def setup(chip):
+    '''Performs routing.'''
+    tool = vivado.tool
```
Comment (gadfort) [inline siliconcompiler/tools/vivado/route.py:5]: couldn't you just import this as `from sc.tools.vivado import tool`

**Review by gadfort — CHANGES_REQUESTED** [review]
Why not split the compile.tcl into the different functions, otherwise you still have this `if task == ` which shouldn't be needed

File: siliconcompiler/tools/vivado/route.py (hunk 2 above)
Comment (nmoroze) [reply siliconcompiler/tools/vivado/route.py:5]: will do, although it's kind of a wash line count wise (I don't like the implicit import of post_process, so I still want to keep the first namespaced import)

**Comment by nmoroze** [comment]
Thanks for the feedback, I refactored `compile.tcl` to use OpenROAD's pattern (which still uses a common entry point and switches on task name). Not sure if you have an opinion on that vs having each task have its own script entry point (which then sources a common pre/post script).

**Comment by gadfort** [comment]
I think it would be cleaner to have each task associate their own script and then source a common startup script.

**Comment by nmoroze** [comment]
Sounds good, done. It does end up adding some boilerplate to grab the refdir of each script.

**Review by gadfort — APPROVED** [review]
Just revert the script breakout

**Pull request merged**